interfaces/ros1/wavemap_ros/app/rosbag_processor.cc

```diff
@@ -75,13 +76,28 @@ int main(int argc, char** argv) {
     rosbag_processor.enableSimulatedClock(nh);
   }
 
+  // Start measuring resource usage
+  ResourceMonitor resource_monitor;
+  resource_monitor.start();
+
   // Process the rosbag
   if (!rosbag_processor.processAll()) {
     return -1;
   }
 
-  wavemap_server.getMap()->prune();
+  // Finish processing the map
   wavemap_server.getPipeline().runOperations(/*force_run_all*/ true);
+  wavemap_server.getMap()->prune();
+
+  // Report the resource usage
+  resource_monitor.stop();
+  LOG(INFO) << "Processing complete.\nResource usage:\n* RAM total: "
```

Comment:
Maybe have a way to directly generate this string for printing if we think it would be shown like this in more than one place.

Reply:
That's a good idea, I'll add a method for this :)